pex/cli/commands/lock.py

```diff
@@ -485,38 +531,41 @@ def _update(self):
                         ),
                         file=output,
                     )
-        if performed_update:
-            original_locked_project_names = {
-                locked_requirement.pin.project_name
-                for locked_resolve in lock_file.locked_resolves
-                for locked_requirement in locked_resolve.locked_requirements
-            }
-            new_requirements = OrderedSet(
-                update
-                for update in updates
-                if update.project_name not in original_locked_project_names
-            )
-            constraints_by_project_name.update(
-                (constraint.project_name, constraint) for constraint in updates
+        if not version_updates:
+            return Ok()
+
+        if dry_run:
+            return Error(exit_code=len(version_updates)) if dry_run is DryRunStyle.CHECK else Ok()
```

Comment:
AFAICT nothing guarantees that the exit code will be <= 255, and if not, I believe the actual exit code will be this number mod 256, which could well be zero. 

Even if not, having an unpredictable exit code seems hard to work with and script around. 

So for those two reasons I think returning a fixed non-zero number makes more sense?

Reply:
True in the 1st. I don't agree on the second. Checking non-zero is straigt-forward. I'll go with the 1st justification, which could lead to issues and fix.